File: 백준/Silver/10845. 큐/큐.js
const input = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n').map(line => line.trim().split(' '));
const [n,...arr] = input;
const arr1 =[]

for(let i=0; i<n; i++){
  switch(arr[i][0]){
    case 'push':
      arr1.push(arr[i][1])
      break

    case 'pop':
      if(arr1.length === 0){
        console.log(-1)
      } else{
        console.log(arr1.shift())
      }
      break

    case 'size':
      console.log(arr1.length)
        break
          
    case 'empty':
      if(arr1.length === 0){
        console.log(1)
      } else{
        console.log(0)
      }
      break

    case 'front':
      if(arr1.length === 0){
        console.log(-1)
      } else{
        console.log(arr1[0])
      }
      break
      
    default:
      if(arr1.length === 0){
        console.log(-1)
      } else{
        console.log(arr1[arr1.length - 1])
      }
  }
}

File: 백준/Silver/10845. 큐/큐.py
from collections import deque
n = int(input())
dct = deque()

for i in range(n):
    arr = list(input().split())
    if arr[0] == 'push':
        dct.append(arr[1])
    elif arr[0] == 'pop':
        if len(dct) == 0:
            print(-1)
        else:
            print(dct.popleft())
    elif arr[0] == 'size':
        print(len(dct))
    elif arr[0] == 'empty':
        if len(dct) == 0:
            print(1)
        else:
            print(0)
    elif arr[0] == 'front':
        if len(dct) == 0:
            print(-1)
        else:
            print(dct[0])
    elif arr[0] == 'back':
        if len(dct) == 0:
            print(-1)
        else:
            print(dct[-1])